exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/FunctionImplementationRegistry.java

```diff
@@ -377,14 +374,12 @@ private ScanResult scan(ClassLoader classLoader, Path path, URL[] urls) throws I
    * Creates local udf directory, if it doesn't exist.
    * Checks if local udf directory is a directory and if current application has write rights on it.
    * Attempts to clean up local udf directory in case jars were left after previous drillbit run.
-   * Local udf directory path is concatenated from drill temporary directory and ${drill.exec.udf.directory.local}.
    *
    * @param config drill config
    * @return path to local udf directory
    */
   private Path getLocalUdfDir(DrillConfig config) {
-    tmpDir = getTmpDir(config);
-    File udfDir = new File(tmpDir, config.getString(ExecConstants.UDF_DIRECTORY_LOCAL));
+    File udfDir = new File(config.getString(ExecConstants.UDF_DIRECTORY_LOCAL));
```

Comment:
Although we talked about using the new system you've implemented here, we have to consider backward compatibility.

Since the original behavior is already visible to users in Drill 1.8, I think we need to leave your original design.

At some point, we'll need to rationalize how Drill handles temp files and storage in DFS. But, until then, your 1.8 design is fine.